tsconfig.json

```diff
@@ -13,12 +13,13 @@
     "noImplicitAny": true,
     "outDir": "./lib",
     "paths": {
-      "*": ["node_modules/*"]
+      "*": ["node_modules/*", "src/*"]
     },
     "resolveJsonModule": true,
     "sourceMap": true,
     "strict": true,
-    "target": "es6"
+    "target": "es6",
+    "lib": ["es6"]
```

Comment:
what does this do?

Reply:
Basically, the build output in `lib/` should be smaller with this as it generates code for more than one target by default.

For the future, can I kindly ask you to read the docs for these kind of questions? https://www.typescriptlang.org/docs/handbook/compiler-options.html

Reply:
Look, I'm perfectly capable to read docs and learn the entirety of TypeScript syntax and best practices. My limitation is time. If I don't understand why a certain change is in as PR and how it supports the core PR goal, then I will just ask. I could have asked a little more clearly, though:

> Why is this change in this PR?

IMO the PR creator has the responsibility to make the review as easy as possible. The harder you make it for reviewers, and the more digging you require them to do, the slower your turnaround time for PRs will be. My main reason for challenging you on these smaller things is that you tend to just throw them into unrelated PRs without any comment or explanation.

This way you're putting more burden on the reviewer to think about
- Why is this change is in this PR?
- Is it required for the core goal of the PR?
- If not, why is it in this PR then?

Especially given that this is our first TypeScript project, and you're the first one that gets to play around with it in detail, IMO you also have a knowledge-sharing and knowledge-building responsibility to proactively disseminate your newly acquired knowledge amongst the team.

Reply:
Don't get me wrong, I'm not saying that you can't throw in a one-liner or two of refactoring into otherwise unrelated PRs. It would be a different story for me if there was a line in the PR description that says:

> ### Refactoring
> Added [`lib`](https://www.typescriptlang.org/docs/handbook/compiler-options.html) parameter to typescript config to improve...

Reply:
Ok, I get your point. Will be more careful with these tiny changes in the future.

From my side:
- It's the 4th PR of this project. I'd appreciate if we can work with the granularity of issues for now and let the ts-setup stuff mature over time. As you know, it's also first time typescript for me, so there's a lot of playing around involved, see `.d.ts` topic. I don't think it's a problem to regard questions 'why is this ts-config param set to true and not false?' not important for now.
- I'm happy to share my knowledge at any time.
- I'd appreciate a bit more autonomy and trust in this project.

Reply:
If you feel mistrusted then I'm sorry. That is definitely not the case. My two main drivers for being so involved in strong-config MRs are:

1) Take part in learning about TypeScript
I unfortunately don't have the time to write a lot of code or read all the TypeScript docs anymore. So for me learning these days often happens in MRs. When I see new ways of doing things. If there was small explanation or little comment then this would go a long way towards knowledge sharing for me :)

2) Setting a high quality bar for PRs.
This is our first real public project, so it heavily contributes to defining our reputation as an engineering team. When I see a PR or issue, I'm always trying to put myself into the shoes of three personas:

- Potential users of the package
   Goes without saying, this is the normal code review process to see whether changes make sense for future users of this package. And I think you're mostly doing a great job here which is why most of my comments have been focussed more on the other two personas.
- Engineers that evaluate joining Brickblock
   Top talent is hard to get. They think twice before joining any company and do their due diligence. If they see messy code or communication in publicly available code or PRs or Issues of a company, that decreases our chances to attract them (obviously none of the code here is messy, but you get the idea)
- External contributors
   If someone creates their first PR for an open source project, they usually look at past PRs as an example for standards around PR description, code quality, communication, etc.